packages/hash/api/src/graphql/typeDefs/knowledge/user.typedef.ts

```diff
@@ -0,0 +1,47 @@
+import { gql } from "apollo-server-express";
+
+export const userTypedef = gql`
+  extend type Query {
+    me(linkResolveDepth: Int!, linkTargetEntityResolveDepth: Int!): Subgraph!
+    """
+    Determines whether a provided shortname is already taken
+    """
+    isShortnameTaken(shortname: String!): Boolean!
+  }
+
+  extend type Mutation {
+    """
+    Create a User.
+    """
+    createUser(
+      """
+      The email address of the user.
+      """
+      emailAddress: String!
+      """
+      The password of the user.
+      """
+      password: String!
+      """
+      Whether or not the user is a HASH instance admin.
+      """
+      isInstanceAdmin: Boolean!
+      """
+      The shortname of the user.
+      """
+      shortname: String
+      """
+      The preferred name of the user.
+      """
+      preferredName: String
+      """
+      The depth of links that are returned in the response subgraph.
+      """
+      linkResolveDepth: Int!
+      """
+      The depth of link target entities that are returned in the response subgraph.
+      """
+      linkTargetEntityResolveDepth: Int!
```

Comment:
I have the same instinct, and have raised this previously. Alfie has had stronger feelings than me about this, and to my understanding prefers these depths having to be set so that they are more explicit. I do worry that without defaults this is something that's going to pollute GQL API query definitions.

cc @Alfred-Mountfield

Reply:
I don't _think_ I argued against defaults (I might have and apologies if I did), but I am definitely against hardcoded values for them for most endpoints.

I think defaults for our _system types_ typedefs are probably sensible, and in some very specific cases hardcoded values might be acceptable.

For other ones I think defaults would be misleading and it would be very hard to pick appropriate ones that work in most general cases.

Reply:
I can't recall in which PR thread I got the wrong impression, apologies Alfie. To be clear you're okay with making these parameters optional, and assigning sensible defaults? In which case my instinct is to go with `0` for now, rather than pick some arbitrary depth?

Reply:
Yep :) Sounds good as long as we're doing it for system types only. And I think `0` is definitely the safest

Reply:
Addressed in https://github.com/hashintel/hash/pull/1360/commits/4f0904f1f1f331972060254fcacf1f69df111294 👍